src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -93,18 +97,34 @@ public bool Equals([AllowNull] List<Section> x, [AllowNull] List<Section> y)
                 DiagnosticSeverity.Warning,
                 isEnabledByDefault: true);
 
+        private readonly static DiagnosticDescriptor MultipleGlobalAnalyzerKeysDescriptor
+            = new DiagnosticDescriptor(
+                "MultipleGlobalAnalyzerKeys",
+                CodeAnalysisResources.WRN_MultipleGlobalAnalyzerKeys_Title,
+                CodeAnalysisResources.WRN_MultipleGlobalAnalyzerKeys,
+                "AnalyzerConfig",
+                DiagnosticSeverity.Warning,
+                isEnabledByDefault: true);
+
         public static AnalyzerConfigSet Create<TList>(TList analyzerConfigs) where TList : IReadOnlyCollection<AnalyzerConfig>
```

Comment:
This method is now essentially hiding diagnostics. Did we consider adding `Obsolete` here or deprecating it through some other mechanism?

Reply:
Yeah, I did think about it. Presumably we could make it public, and pass the diagnostics out (DiagnosticBag isn't public). But my thinking ended up being we would just expose it if anyone ended up needing it. Happy to refactor to do it upfront though,

---
In reply to: [423293758](https://github.com/dotnet/roslyn/pull/43889#discussion_r423293758) [](ancestors = 423293758)

Reply:
@333fred for guidance

Reply:
I think it's probably worth creating creating an overload that returns diagnostics from merging as an out parameter. I'm not 100% sure that we should obsolete this one though. In fact, it seems odd to me that there would be a warning for multiple sets, as opposed to a hidden-by-default diagnostic that a user can turn to a warning/error if needed. You might depend on an analyzer that configs one of its defaults to be on/warning, but another analyzer you depend on says "My analyzer in this package supersedes that one. You still want the original package for the other analyzers, but I'm going to provide a config to turn that specific one off."

Reply:
Cool, I have a commit that does exactly that. The trouble is because these are global configs they are all considered to have equal precedence, and thus can't supersede each other. So the only thing we can really do is warn and unset.